application-admintools-default/src/main/java/com/xwiki/admintools/internal/usage/InstanceUsageManager.java

```diff
@@ -148,106 +132,76 @@ public String renderTemplate()
-                            long attachmentsSize = wikiData.getAttachmentsSize();
-                            long lowerBound = Long.parseLong(interval[0]);
-                            long upperBound = "x".equals(interval[1]) ? Long.MAX_VALUE : Long.parseLong(interval[1]);
-                            return attachmentsSize > lowerBound && attachmentsSize < upperBound;
-                        case ATTACHMENTS_COUNT_KEY:
-                            return wikiData.getAttachmentsCount().equals(Long.parseLong(filter.getValue()));
-                        case DOCUMENTS_COUNT_KEY:
-                            return wikiData.getDocumentsCount().equals(Long.parseLong(filter.getValue()));
-                        default:
-                            throw new IllegalArgumentException("Invalid filter field: " + filter.getKey());
-                    }
-                });
+        try {
+            Collection<WikiDescriptor> wikisDescriptors = getFilteredWikis(filters);
+            return recycleBinsProvider.getWikisRecycleBinsSize(wikisDescriptors, filters, sortColumn, order);
+        } catch (Exception e) {
+            logger.warn("There have been issues while gathering wikis recycle bins data. Root cause is: [{}]",
+                ExceptionUtils.getRootCauseMessage(e));
+            throw new RuntimeException(e);
+        }
     }
 
-    private void applySort(List<WikiSizeResult> list, String sort, String order)
+    private Collection<WikiDescriptor> getFilteredWikis(Map<String, String> filters) throws WikiManagerException
     {
-        Comparator<WikiSizeResult> comparator = null;
-        switch (sort) {
-            case NAME_KEY:
-                comparator = Comparator.comparing(WikiSizeResult::getName);
-                break;
-            case USER_COUNT_KEY:
-                comparator = Comparator.comparing(WikiSizeResult::getUserCount);
-                break;
-            case ATTACHMENTS_SIZE_KEY:
-                comparator = Comparator.comparing(WikiSizeResult::getAttachmentsSize);
-                break;
-            case ATTACHMENTS_COUNT_KEY:
-                comparator = Comparator.comparing(WikiSizeResult::getAttachmentsCount);
-                break;
-            case DOCUMENTS_COUNT_KEY:
-                comparator = Comparator.comparing(WikiSizeResult::getDocumentsCount);
-                break;
-            default:
-                break;
-        }
-        if (comparator != null) {
-            if ("desc".equals(order)) {
-                comparator = comparator.reversed();
-            }
-            list.sort(comparator);
+        Collection<WikiDescriptor> wikisDescriptors = this.wikiDescriptorManager.getAll();
+        String filteredName = filters.get(WIKI_NAME_KEY);
+        if (filteredName != null && !filteredName.isEmpty()) {
+            wikisDescriptors.removeIf(wiki -> !wiki.getPrettyName().toLowerCase().contains(filteredName.toLowerCase()));
+            filters.remove(WIKI_NAME_KEY);
```

Comment:
Why you remove it after?

Reply:
I remove it so that it does not iterate in the `checkFilters` function, as the remaining wikis are already satisfying the name filter. I find it to be one more useless iteration if left.